src/chat/chatReducers.js

```diff
@@ -95,19 +95,26 @@ export default (state: ChatState = initialState, action: Action) => {
           x => !(x.emoji_name === action.emoji && x.user.email === action.user.email),
         ),
       }));
-
     case EVENT_NEW_MESSAGE: {
-      return {
+      let stateChange = false;
+      const newState = {
         ...state,
         messages: Object.keys(state.messages).reduce((msg, key) => {
           const isInNarrow = isMessageInNarrow(action.message, JSON.parse(key), action.ownEmail);
-          msg[key] = isInNarrow ? [...state.messages[key], action.message] : state.messages[key];
+          const isMessageAlreadySaved = isInNarrow
+            ? state.messages[key].find(item => action.message.id === item.id) !== undefined
```

Comment:
I refactored the code 

EVENT_NEW_MESSAGE is very different from MESSAGE_FETCH_SUCCESS, hence did not take code from there!